roles/fio_distributed/templates/server_vm.yml.j2

```diff
@@ -22,60 +22,70 @@ spec:
       cores: {{ workload_args.vm_cores | default(1) }}
     devices:
       disks:
-        - name: registrydisk
-        - name: cloudinitdisk
+        - disk:
+            bus: {{ workload_args.vm_bus | default('virtio') }}
+          name: registrydisk
+        - disk:
+            bus: {{ workload_args.vm_bus | default('virtio') }}
+          name: cloudinitdisk
 {% if workload_args.storageclass is defined
     or workload_args.hostpath is defined %}
         - disk:
-            bus: virtio
+            bus: {{ workload_args.vm_bus | default('virtio') }}
           name: data-volume
           serial: data
+{% else %}
+        - disk:
+            bus: {{ workload_args.vm_bus | default('virtio') }}
+          name: emptydisk
 {% endif %}
     resources:
       requests:
         memory: {{ workload_args.vm_memory | default('5G') }}
   volumes:
     - name: registrydisk
       containerDisk:
-        image: {{ workload_args.vm_image | default('kubevirt/fedora-cloud-container-disk-demo:latest') }}
+        image: {{ workload_args.vm_image | default('quay.io/kubevirt/fedora-cloud-container-disk-demo:latest') }}
     - name: cloudinitdisk
       cloudInitNoCloud:
         userData: |-
           #cloud-config
-          password: fedora
+          password: ripsaw
           chpasswd: { expire: False }
           bootcmd:
+            - "mkdir -p {{ fio_path }} || true"
 {% if workload_args.storageclass is defined
     or workload_args.hostpath is defined %}
-            - "mkdir -p {{ fio_path }} || true"
-            - mkfs.ext4 /dev/disk/by-id/virtio-data
-            - "mount /dev/disk/by-id/virtio-data {{ fio_path }}"
+            - "[ -e /dev/disk/by-id/*data ] && disk=$(basename `ls /dev/disk/by-id/*data`) && mkfs.ext4 /dev/disk/by-id/$disk && mount /dev/disk/by-id/$disk {{ fio_path }}"
```

Comment:
Better to write

$(shopt -s nullglob; basename /dev/disk/by-id/*data)

Note that if you want to use ls like this that $(...) constructs nest; there's no need for the backquote construct.

Reply:
Thanks for the tip, works well. Also I found I can simplify the emptydisk fallback by adding the 'data' label which makes the code cleaner, but my testing is a bit delayed due to a wrapper issue ([PR376](https://github.com/cloud-bulldozer/benchmark-wrapper/pull/376) should fix it)